sysl2/sysl/seqs.go

```diff
@@ -0,0 +1,288 @@
+		}
+		initialStr = strings.Replace(initialStr, item, value, 1)
+	}
+
+	return seqs.RemovePercentSymbol(initialStr)
+}
+
+func constructSimpleParser(former, latter string) *SimpleParser {
+	fmtstr := former
+	if former == "" {
+		fmtstr = latter
+	}
+
+	return &SimpleParser{self: fmtstr}
+}
+
+func DoConstructSequenceDiagrams(root_model, endpoint_format, app_format, title, plantuml, output string,
+	endpoints, apps, modules []string, blackboxes [][]string) {
+	mod := loadApp(root_model, modules)
+	syslSdFilters, exists := os.LookupEnv("SYSL_SD_FILTERS")
+	epFilters := []string{}
+	if exists {
+		epFilters = append(epFilters, strings.Split(syslSdFilters, ",")...)
+	} else {
+		epFilters = append(epFilters, "*")
+	}
+
+	if strings.Contains(output, "%(epname)") {
+		spout := &SimpleParser{self: output}
+		for _, appName := range apps {
+			app := mod.Apps[appName]
+			bbs := seqs.TransformBlackBoxes(app.GetAttrs()["blackboxes"].GetA().GetElt())
+			spseqtitle := constructSimpleParser(app.GetAttrs()["seqtitle"].GetS(), title)
+			spep := constructSimpleParser(app.GetAttrs()["epfmt"].GetS(), endpoint_format)
+			spapp := constructSimpleParser(app.GetAttrs()["appfmt"].GetS(), app_format)
+			keys := []string{}
+			for k := range app.GetEndpoints() {
+				keys = append(keys, k)
+			}
+			sort.Strings(keys)
+			for _, k := range keys {
+				epAttrs := app.GetEndpoints()[k].GetAttrs()
+				output_dir := spout.fmtOutput(appName, k, app.GetEndpoints()[k].GetLongName(), epAttrs)
+				bbs2 := seqs.TransformBlackBoxes(app.GetEndpoints()[k].GetAttrs()["blackboxes"].GetA().GetElt())
+				varrefs := seqs.MergeAttributes(app.GetAttrs(), app.GetEndpoints()[k].GetAttrs())
+				sdEndpoints := []string{}
+				statements := app.GetEndpoints()[k].GetStmt()
+				for _, stmt := range statements {
+					parts := stmt.GetCall().GetTarget().GetPart()
+					ep := stmt.GetCall().GetEndpoint()
+					sdEndpoints = append(sdEndpoints, strings.Join(parts, " :: ")+" <- "+ep)
+				}
+
+				sd := &sequenceDiagParam{
+					endpoints:   sdEndpoints,
+					AppLabeler:  spapp,
+					EndpointLabeler: spep,
+					title:       spseqtitle.fmtSeq(app.GetEndpoints()[k].GetName(), app.GetEndpoints()[k].GetLongName(), varrefs),
+					blackboxes:  append(bbs, bbs2...),
+				}
+				out, _ := generateSequenceDiag(mod, sd)
+				seqs.OutputPlantuml(output_dir, plantuml, out)
+			}
+		}
+	} else {
+		if endpoints == nil {
+			return
+		}
+		spep := constructSimpleParser("", endpoint_format)
+		spapp := constructSimpleParser("", app_format)
+		sd := &sequenceDiagParam{
+			endpoints:   endpoints,
+			AppLabeler:  spapp,
+			EndpointLabeler: spep,
+			title:       title,
+			blackboxes:  blackboxes,
+		}
+		out, _ := generateSequenceDiag(mod, sd)
+		seqs.OutputPlantuml(output, plantuml, out)
+	}
+}
+
+// DoGenerateSequenceDiagrams generate sequence diagrams for the given model
+func DoGenerateSequenceDiagrams(stdout, stderr io.Writer, flags *flag.FlagSet, args []string) int {
```

Comment:
#210

Reply:
Maybe we can create a new PR for the #210

Reply:
Agreed. The presence of an issue number generally impies, "Fix this later!"